roms_tools/setup/initial_conditions.py

```diff
@@ -746,148 +714,34 @@ def plot(
             with ProgressBar():
                 self.ds[var_name].load()
 
-        # Retrieve depth coordinates
-        if s is not None:
-            layer_contours = False
-        # Note that `layer_depth_{loc}` has already been computed during `__post_init__`.
-        layer_depth = self.ds_depth_coords[f"layer_depth_{loc}"].squeeze().load()
-
-        # Slice the field as desired
-        def _slice_and_assign(
-            field,
-            mask,
-            layer_depth,
-            title,
-            dim_name,
-            dim_values,
-            idx,
-        ):
-            if dim_name in field.dims:
-                title = title + f", {dim_name} = {dim_values[idx].item()}"
-                field = field.isel(**{dim_name: idx})
-                mask = mask.isel(**{dim_name: idx})
-                layer_depth = layer_depth.isel(**{dim_name: idx})
-                if "s_rho" in field.dims:
-                    field = field.assign_coords({"layer_depth": layer_depth})
-            else:
-                raise ValueError(
-                    f"None of the expected dimensions ({dim_name}) found in field."
-                )
-            return field, mask, layer_depth, title
-
-        title = field.long_name
-        if s is not None:
-            title = title + f", s_rho = {field.s_rho[s].item()}"
-            field = field.isel(s_rho=s)
-            layer_depth = layer_depth.isel(s_rho=s)
-            field = field.assign_coords({"layer_depth": layer_depth})
+        if self.adjust_depth_for_sea_surface_height:
+            zeta = self.ds.zeta.squeeze().load()
         else:
-            depth_contours = False
-
-        if eta is not None:
-            field, mask, layer_depth, title = _slice_and_assign(
-                field,
-                mask,
-                layer_depth,
-                title,
-                "eta_rho" if "eta_rho" in field.dims else "eta_v",
-                field.eta_rho if "eta_rho" in field.dims else field.eta_v,
-                eta,
-            )
+            zeta = 0
 
-        if xi is not None:
-            field, mask, layer_depth, title = _slice_and_assign(
-                field,
-                mask,
-                layer_depth,
-                title,
-                "xi_rho" if "xi_rho" in field.dims else "xi_u",
-                field.xi_rho if "xi_rho" in field.dims else field.xi_u,
-                xi,
-            )
+        field = self.ds[var_name].squeeze()
 
         # Choose colorbar
```

Comment:
Has this comment become obsolete?  The trailing lines determine the color-map, not a color bar. Can it be removed?